### PR TITLE
Fix the need to redeclare viewsets using PEP0562

### PR DESCRIPTION
**Pull request by rochacbruno**
No-Issue

### CODE REVIEW

File: galaxy_ng/app/viewsets.py [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
+
+    This file is necessary to prevent the DRF web API browser from breaking on all of the
+    pulp/api/v3/repositories/ endpoints.
+    Pulp expects models it manages to come with a viewset attached to them. Since the
+    ContainerRegistryRemote model is a pulp remote, pulp expects it to have a pulp
+    viewset attached to it. Pulp associates viewsets with models by looking in
+    <plugin_name>.app.viewsets. Since galaxy_ng stores it's viewsets in a different
+    module, this file is necesary for pulp tbe able to associate the ContainerRegistryRemote
+    model to a viewset.
+    Without this viewset defined here, pulp get's confused when it tries to auto generate
```
Comment (daviddavis) [inline galaxy_ng/app/viewsets.py:25]:
```suggestion
    Without this viewset defined here, pulp gets confused when it tries to auto generate
```

File: galaxy_ng/app/viewsets.py [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
+    the form on the repositories endpoint because it tries to provide a dropdown
+    on the remote field and can't find a viewset name for galaxy's ContainerRegistryRemote model.
+
+    """
+
+    if name in viewsets:
+        try:
+            module = __import__(viewsets[name])
+        except ImportError as e:
+            log.error("Error importing viewset: %s" % e)
```
Comment (daviddavis) [inline galaxy_ng/app/viewsets.py:35]: Maybe worth being consistent with line 39?

File: galaxy_ng/app/viewsets.py [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
+
+
+def __getattr__(self, name):
+    """
+    This is a hack to get pulp to associate the model with a viewset.
+
+    This file is necessary to prevent the DRF web API browser from breaking on all of the
+    pulp/api/v3/repositories/ endpoints.
+    Pulp expects models it manages to come with a viewset attached to them. Since the
+    ContainerRegistryRemote model is a pulp remote, pulp expects it to have a pulp
```
Comment (daviddavis) [inline galaxy_ng/app/viewsets.py:20]:
```suggestion
    ContainerRegistryRemote model is a pulp remote, pulp expects it to have a NamedModelViewSet
```

File: galaxy_ng/app/viewsets.py [hunk 4]
```diff
@@ -1,35 +1,40 @@
...
+    viewset attached to it. Pulp associates viewsets with models by looking in
+    <plugin_name>.app.viewsets. Since galaxy_ng stores it's viewsets in a different
```
Comment (daviddavis) [inline galaxy_ng/app/viewsets.py:22]:
```suggestion
    <plugin_name>.app.viewsets. Since galaxy_ng stores its viewsets in a different
```

**Review by daviddavis — APPROVED** [review]
LGTM. A couple small suggestions.

File: galaxy_ng/app/viewsets.py (hunk 2 above)
Comment (rochacbruno) [reply galaxy_ng/app/viewsets.py:35]: In this case the viewset is declared but there was an error importing it, in the other case it is really not found.

Comment (daviddavis) [reply galaxy_ng/app/viewsets.py:35]: Oh I was just wondering about the use of %

**Review by newswangerd — APPROVED** [review]

**Pull request merged**